File: sortedSquared.js
const squaredArray = (array) => {
    // let sortedSquare = new Array(array.length).fill(0);
    // for (let index = 0; index < array.length; index++) {
    //     value = array[index];
    //     sortedSquare[index] = value + value;
    // Array.prototype.sort(sortedSquare);
    // return sortedSquare;
    // }
    let sortedSquare = array.map(num => num * num); // Calculate squares of elements
    sortedSquare.sort((a, b) => a - b); // Sort the squared elements in ascending order
    return sortedSquare;
}
// O(nlogn)

let array = [-4, -2, 1, 3, 9]
console.log(squaredArray(array))

File: sortedSquared.cpp
#include<bits/stdc++.h>
using namespace std;

vector<int> squaredArray(vector<int>& array){
    vector<int> squared;
    for (int num : array) {
        squared.push_back(num * num); // Calculate squares
    }

    std::sort(squared.begin(), squared.end()); // Sort squared values

    return squared;
}

int main() {
    std::vector<int> array = {-4, -1, 0, 3, 10}; // Example array
    std::vector<int> sortedSquare = squaredArray(array);

    for (int val : sortedSquare) {
        std::cout << val << " ";
    }
    // Output: 0 1 9 16 100

    return 0;
}